src/sdk/namespace_cache.js

```diff
@@ -157,18 +168,27 @@ class NamespaceCache {
     /////////////////
 
     async list_objects(params, object_sdk) {
-        // TODO listing from cache only for deevelopment
-        return this.namespace_nb.list_objects(params, object_sdk);
+        const get_from_cache = params.get_from_cache;
+        params = _.omit(params, 'get_from_cache');
+        if (get_from_cache) {
+            return this.namespace_nb.list_objects(params, object_sdk).then(res => {
+                _.forEach(res.objects, obj => {
+                    obj.create_time = obj.last_modified_time ? obj.last_modified_time : obj.create_time;
+                });
+                return res;
+            });
+        }
+        return this.namespace_hub.list_objects(params, object_sdk);
     }
 
     async list_uploads(params, object_sdk) {
-        // TODO listing from cache only for deevelopment
-        return this.namespace_nb.list_uploads(params, object_sdk);
+
+        return this.namespace_hub.list_uploads(params, object_sdk);
     }
 
     async list_object_versions(params, object_sdk) {
-        // TODO listing from cache only for deevelopment
-        return this.namespace_nb.list_object_versions(params, object_sdk);
+        // Always return latest version.
+        return this.list_objects(params, object_sdk);
```

Comment:
So this means that even if we have versions in the hub we do not list those.
I don't think this is what we should be doing.
I think we should list version from hub, and just avoid caching versions other than latest on GET.

Reply:
The `list_objects` function will return from the hub or from the cache based on the `get_from_cache` param value. So this should return from the hub if there are versions on the hub. I have updated the comment

Reply:
ok